common/rpc/rpc.go

```diff
@@ -189,12 +190,20 @@ func getListenIP(cfg *config.RPC, logger log.Logger) net.IP {
 // CreateRemoteFrontendGRPCConnection creates connection for gRPC calls
 func (d *RPCFactory) CreateRemoteFrontendGRPCConnection(rpcAddress string) *grpc.ClientConn {
 	var tlsClientConfig *tls.Config
-	var err error
 	if d.tlsFactory != nil {
-		hostname, _, err2 := net.SplitHostPort(rpcAddress)
-		if err2 != nil {
-			d.logger.Fatal("Invalid rpcAddress for remote cluster", tag.Error(err2))
+		target := rpcAddress
+		hostname := ""
+
+		// Try parsing the address as a URI
+		if u, parseErr := url.Parse(rpcAddress); parseErr == nil && u.Scheme != "" && strings.Contains(rpcAddress, "://") {
+			target = strings.TrimPrefix(u.Path, "/") // remove any leading slashes
+		}
```

Comment:
I would rather only try to parse if needed and avoid swallowing errors if they are expected.

```suggestion
		// Try parsing the address as a URI to support custom gRPC resolvers.
		if strings.Contains(rpcAddress, "://") {
			u, parseErr := url.Parse(rpcAddress)
			if parseErr != nil {
				d.logger.Fatal("Invalid rpcAddress for remote cluster. Unable to parse URL", tag.Error(parseErr), tag.String("rpcAddress", rpcAddress))
			}
			// Use the URL's path as the target as expected by gRPC resolvers such "passthrough".
			target = strings.TrimPrefix(u.Path, "/") // remove any leading slashes
		}
```